### PR TITLE
Fixed #28992 -- added validate into GenericIPAddressField's prep value

### PR DESCRIPTION
**Pull request by cansarigol**
Ticket: [#28992](https://code.djangoproject.com/ticket/28992)

### CODE REVIEW

File: tests/model_fields/test_genericipaddressfield.py [hunk 1]
```diff
@@ -38,6 +42,6 @@ def test_blank_string_saved_as_null(self):
         self.assertIsNone(o.ip)
 
     def test_save_load(self):
-        instance = GenericIPAddress.objects.create(ip='::1')
+        instance = GenericIPAddress.objects.create(ip='1.1.1.1')
```
Comment (claudep) [inline tests/model_fields/test_genericipaddressfield.py]: Could you explain this change?

Comment (cansarigol) [reply tests/model_fields/test_genericipaddressfield.py]: GenericIPAddress ip field's protocol=ipv4 because of that ::1 value is not valid

Comment (claudep) [reply tests/model_fields/test_genericipaddressfield.py]: Ah, yes, obviously!

**Comment by carltongibson** [comment]
It's nearly my lunchtime so 🤯 a bit but... I'm not seeing quite how this solve the reported issue, which was a 500 from `search_fields` in the admin. 

* Won't raising the `ValueError` still result in the 500 error response? (Just a different exception.)
* Shouldn't the submitted `search_fields` values be validated (using a form) similar to your other PR #10895 @cansarigol, which I see we still need to get finished? (I'll pick that up again... — sorry for the delay there. 🤹‍♀️)

**Comment by cansarigol** [comment]
you're right about admin side. I will look asap. I wanted to take an approved about which exception will be raised for field-side from you.

**Comment by carltongibson** [comment]
It’s not immediately clear to me that we get that much benefit raising the ValueError here. I guess it’s quicker than a trip to the DB...

Thinking about this, it seems we’ll need to expose the form, via the ChangeList class..., so users can add their own validators. We need to have a little think about how that’ll look, but first thing is to get #10895 in.

**Pull request closed without merging**